beacon-chain/rpc/lookup/blocker.go

```diff
@@ -215,54 +214,63 @@ func (p *BeaconDbBlocker) Blobs(ctx context.Context, id string, indices []uint64
+		return nil, &core.RpcError{Err: fmt.Errorf("block %#x not found in db", root), Reason: core.NotFound}
 	}
 	b, err := p.BeaconDB.Block(ctx, bytesutil.ToBytes32(root))
 	if err != nil {
-		return nil, &core.RpcError{Err: errors.Wrap(err, "failed to retrieve block from db"), Reason: core.Internal}
+		return nil, &core.RpcError{Err: errors.Wrapf(err, "failed to retrieve block %#x from db", root), Reason: core.Internal}
 	}
-	// if block is not in the retention window  return 200 w/ empty list
+
+	// if block is not in the retention window, return 200 w/ empty list
 	if !p.BlobStorage.WithinRetentionPeriod(slots.ToEpoch(b.Block().Slot()), slots.ToEpoch(p.GenesisTimeFetcher.CurrentSlot())) {
 		return make([]*blocks.VerifiedROBlob, 0), nil
 	}
+
 	commitments, err := b.Block().Body().BlobKzgCommitments()
 	if err != nil {
-		return nil, &core.RpcError{Err: errors.Wrap(err, "failed to retrieve kzg commitments from block"), Reason: core.Internal}
+		return nil, &core.RpcError{Err: errors.Wrapf(err, "failed to retrieve kzg commitments from block %#x", root), Reason: core.Internal}
 	}
 	// if there are no commitments return 200 w/ empty list
 	if len(commitments) == 0 {
 		return make([]*blocks.VerifiedROBlob, 0), nil
 	}
-	if len(indices) == 0 {
-		m, err := p.BlobStorage.Indices(bytesutil.ToBytes32(root), b.Block().Slot())
-		if err != nil {
-			log.WithFields(log.Fields{
-				"blockRoot": hexutil.Encode(root),
-			}).Error(errors.Wrapf(err, "could not retrieve blob indices for root %#x", root))
-			return nil, &core.RpcError{Err: fmt.Errorf("could not retrieve blob indices for root %#x", root), Reason: core.Internal}
-		}
-		for k, v := range m {
-			if v {
-				if k >= len(commitments) {
-					return nil, &core.RpcError{Err: fmt.Errorf("blob index %d is more than blob kzg commitments :%dd", k, len(commitments)), Reason: core.BadRequest}
+
+	m, err := p.BlobStorage.Indices(bytesutil.ToBytes32(root), b.Block().Slot())
+	if err != nil {
+		return nil, &core.RpcError{Err: fmt.Errorf("could not retrieve blob indices for block %#x", root), Reason: core.Internal}
+	}
+
+	indicesRequested := len(indices) > 0
+	for k, v := range m {
+		if v {
+			if k >= len(commitments) {
+				return nil, &core.RpcError{
+					Err:    fmt.Errorf("blob index %d is more than blob kzg commitment count %d for block %#x", k, len(commitments), root),
+					Reason: core.Internal,
 				}
+			}
+			if !indicesRequested {
 				indices = append(indices, uint64(k))
 			}
+		} else if indicesRequested {
+			for _, ix := range indices {
+				if uint64(k) == ix {
+					return nil, &core.RpcError{Err: fmt.Errorf("no blob at index %d found for block %#x", k, root), Reason: core.NotFound}
+				}
+			}
 		}
 	}
-	// returns empty slice if there are no indices
+
 	blobs := make([]*blocks.VerifiedROBlob, len(indices))
 	for i, index := range indices {
 		vblob, err := p.BlobStorage.Get(bytesutil.ToBytes32(root), index)
 		if err != nil {
-			log.WithFields(log.Fields{
```

Comment:
I removed this log and one more log because I don't see the point of logging out one specific thing from the whole function. As a matter of fact, I don't think we log out anything else from the whole Beacon API (note that this file is in `beacon-chain/rpc`, so it's limited to APIs). In my opinion if we find such logs useful, we should write a middleware that will log out every error from `httputil.HandleError` (most likely on the `Debug` level)